WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -373,6 +373,16 @@ public void onEventMainThread(PostEvents.PostUploadStarted event) {
         }
     }
 
+    /*
+    * Upload cancelled (probably due to failed media), reload so correct status on uploading post appears
+    */
+    @SuppressWarnings("unused")
+    public void onEventMainThread(PostEvents.PostUploadCanceled event) {
+        if (isAdded() && mSite.getId() == event.localSiteId) {
```

Comment:
I understand `mSite` shouldn't be null, but reading the `onCreate` sequence it seems this class is registered for FLuxC event listening _before_ the `mSite` property is initialized in `updateSiteOrFinishActivity()`. While that might deserve further analysis, I'd say at least for the sake of this PR we check `mSite` is different than `null` here to avoid a NPE.